File: tws-graph/src/tws_graph/indexer/extractors/rust.py
"""Rust extractor — BaseExtractor subclass."""

from ..base import BaseExtractor, ExtractionContext


class RustExtractor(BaseExtractor):
    extensions = [".rs"]
    tree_sitter_languages = ["rust"]
    language_name = "rust"

    def extract(self, source: bytes, tree, ctx: ExtractionContext) -> None:
        from ..rust_extractor import visit_rust

        result = visit_rust(ctx.file_path, source.decode("utf-8", errors="replace"), tree)

        ctx.result.nodes.extend(result.nodes)
        ctx.result.edges.extend(result.edges)
        ctx.result.errors.extend(result.errors)

File: tws-graph/rust_core/src/indexer/extractors/rust.rs
//! rust language extractor.

use crate::indexer::context::ExtractionContext;
use crate::traits::Extractor;
use tree_sitter::Tree;

pub struct RustExtractor;

impl Extractor for RustExtractor {
    fn extensions(&self) -> Vec<&'static str> {
        vec!["rs"]
    }
    fn languages(&self) -> Vec<&'static str> {
        vec!["rust"]
    }
    fn extract(
        &self,
        _source: &[u8],
        _tree: &Tree,
        _ctx: &mut ExtractionContext,
    ) -> anyhow::Result<()> {
        Ok(())
    }
}
